### PR TITLE
MTDSA-23359 Multi-schema support for Historic UK Non-FHL Annual endpoints

### PR DESCRIPTION
**Pull request by mattstephens-hmrc**

### CODE REVIEW

File: app/v4/retrieveHistoricNonFhlUkPropertyAnnualSubmission/RetrieveHistoricNonFhlUkPropertyAnnualSubmissionConnector.scala [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+    request match {
+      case def1: Def1_RetrieveHistoricNonFhlUkPropertyAnnualSubmissionRequestData =>
+        import def1._
+
+        val downstreamUri =
+          IfsUri[Def1_RetrieveHistoricNonFhlUkPropertyAnnualSubmissionResponse](
+            s"income-tax/nino/$nino/uk-properties/other/annual-summaries/${taxYear.asDownstream}")
+
+        val response = get(downstreamUri)
```
Comment (LouiseWong98) [inline app/v4/retrieveHistoricNonFhlUkPropertyAnnualSubmission/RetrieveHistoricNonFhlUkPropertyAnnualSubmissionConnector.scala:53]: can shorten this to just get(downstreamUri)

Comment (mattstephens-hmrc) [reply app/v4/retrieveHistoricNonFhlUkPropertyAnnualSubmission/RetrieveHistoricNonFhlUkPropertyAnnualSubmissionConnector.scala:53]: Annoyingly, it's needed to keep the Scala compiler happy
